### PR TITLE
Add a new category 'asic_gen' in dut basic facts.

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

Test cases under the tests/sub_port_interfaces only support td2 asic generation, we add a new category 'asic_gen' in dut basic facts to skip unsupported asic in advance. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Test cases under the tests/sub_port_interfaces only support td2 asic generation, we add a new category 'asic_gen' in dut basic facts to skip unsupported asic in advance. 

#### How did you do it?
Get the asic generation infomation from  `ansible/group_vars/sonic/variables`,  and add it in dut basic facts. Also,  modify the conditions in `tests_mark_conditions.yaml` to skip cases. 

#### How did you verify/test it?
Running test cases on support/unsupport asic.

### CODE REVIEW

File: tests/common/plugins/conditional_mark/__init__.py [hunk 1]
```diff
@@ -85,6 +85,25 @@ def load_conditions(session):
 
     return conditions_list
 
+def read_asic_name(hwsku):
+    asic_name_file = os.path.dirname(__file__) + ASIC_NAME_PATH
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/__init__.py:89]: Is the `ASIC_NAME_PATH` defined somewhere?

File: tests/common/plugins/conditional_mark/__init__.py [hunk 2]
```diff
@@ -85,6 +85,25 @@ def load_conditions(session):
 
     return conditions_list
 
+def read_asic_name(hwsku):
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/__init__.py:88]: Add some comments here as documentation?

Also update the README.md file for this plugin?

File: tests/common/plugins/conditional_mark/__init__.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply tests/common/plugins/conditional_mark/__init__.py:89]: oh, sorry, I forget to paste this.

File: tests/common/plugins/conditional_mark/__init__.py (hunk 2 above)
Comment (yutongzhang-microsoft) [reply tests/common/plugins/conditional_mark/__init__.py:89]: done

File: tests/common/plugins/conditional_mark/__init__.py [hunk 3]
```diff
@@ -85,6 +86,35 @@ def load_conditions(session):
...
+
+        for key, value in asic_name.items():
+            if ('td' not in key) and ('th' not in key):
+                asic_name.pop(key)
+
+        for name, hw in asic_name.items():
+            if hwsku in hw:
+                return name.split('_')[1]
+
+        return None
```
Comment (ZhaohuiS) [inline tests/common/plugins/conditional_mark/__init__.py:114]: Is it better to return "unknown" such as the function `get_asic_name` does in `sonic.py`?

Comment (yutongzhang-microsoft) [reply tests/common/plugins/conditional_mark/__init__.py:114]: It will be better.

**Review by ZhaohuiS — APPROVED** [review]

**Pull request merged**